drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,668 @@
+/*
+ * Copyright (C) ??? 2023 
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx126x
+ * @{
+ * @file
+ * @brief       Radio HAL implementation for the Semtech SX126x 
+ *
+ * @author      Klim Evdokimov <klimevdokimov@mail.ru>
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#include "net/ieee802154/radio.h"
+#include "event/callback.h"
+#include "event/thread.h"
+#include "ztimer.h"
+
+#include "kernel_defines.h"
+#include "event.h"
+
+#include "sx126x.h"
+#include "sx126x_params.h"
+#include "sx126x_internal.h"
+
+#define LORA_ACK_REPLY_US          1024
```

Comment:
to align it with IEEE 802.15.4, this should be 54 symbols.
If using SF7BW125, this means 54 * 1024[us] = 55.3 [ms]